include/swift/Basic/SourceManager.h

```diff
@@ -231,8 +231,9 @@ class SourceManager {
   void verifyAllBuffers() const;
 
   /// Translate line and column pair to the offset.
+  /// If the column number isn't given, return the offset of the end of the line.
```

Comment:
This is rather surprising to me, if I see a `resolveFromLineCol(bufID, line)` call somewhere I will intuitively think that it is getting the offset at the beginning of the passed `line`, not its end. How about:
* Either at the call site where you want the offset of end-of-line, call `resolveFromLineCol(bufID, line+1, 1)` and subtract 1 from the given offset, or maybe just use the beginning offset of the next line directly for comparison,
* or add a new function `resolveOffsetForEndOfLine` to make it more readable.